### PR TITLE
Ensure no duplicate functions because of lowercase naming

### PR DESCRIPTION
**Pull request by ed-holland**
issue #3633

### CODE REVIEW

File: src/objects/zcl_abapgit_object_fugr.clas.abap [hunk 1]
```diff
@@ -518,6 +518,12 @@ CLASS ZCL_ABAPGIT_OBJECT_FUGR IMPLEMENTATION.
       zcx_abapgit_exception=>raise( 'Error from RS_FUNCTION_POOL_CONTENTS' ).
     ENDIF.
 
+* The result can also contain function which are lowercase.
+    LOOP AT rt_functab INTO ls_functab.
```
Comment (bigld) [inline src/objects/zcl_abapgit_object_fugr.clas.abap:522]: I would prefer a field symbol?
`LOOP AT rt_functab ASSIGNING <ls_functab>.`

File: src/objects/zcl_abapgit_object_enhs.clas.abap [hunk 2]
```diff
@@ -185,14 +246,37 @@ CLASS ZCL_ABAPGIT_OBJECT_ENHS IMPLEMENTATION.
...
+    DATA lx_locked TYPE REF TO cx_enh_is_locked.
 
     lv_spot_name = ms_item-obj_name.
 
     TRY.
         li_spot_ref = cl_enh_factory=>get_enhancement_spot( lv_spot_name ).
 
-      CATCH cx_enh_root INTO lx_root.
-        zcx_abapgit_exception=>raise( 'Error from CL_ENH_FACTORY' ).
+      CATCH cx_enh_io_error INTO lx_io_error.
```
Comment (bigld) [inline src/objects/zcl_abapgit_object_enhs.clas.abap:260]: maybe we can use get_text / get_longtext of the standard exception?

File: src/objects/zcl_abapgit_object_enhs.clas.abap [hunk 3]
```diff
@@ -70,8 +77,34 @@ CLASS ZCL_ABAPGIT_OBJECT_ENHS IMPLEMENTATION.
 
         li_enh_object->unlock( ).
 
-      CATCH cx_enh_root.
-        zcx_abapgit_exception=>raise( 'Error from CL_ENH_FACTORY' ).
+      CATCH cx_enh_io_error INTO lx_io_error.
```
Comment (bigld) [inline src/objects/zcl_abapgit_object_enhs.clas.abap:80]: maybe we can use get_text / get_longtext of the standard exception?

**Comment by ed-holland** [comment]
Replaced with PR #3652

**Pull request closed without merging**